flink-end-to-end-tests/flink-streaming-kinesis-test/src/main/java/org/apache/flink/streaming/kinesis/test/KinesisTableApiITCase.java

```diff
@@ -0,0 +1,162 @@
+        System.setProperty("com.amazonaws.sdk.disableCertChecking", "1");
+        System.setProperty("com.amazonaws.sdk.disableCbor", "1");
+        System.setProperty(
+                "org.apache.flink.kinesis.shaded.com.amazonaws.sdk.disableCertChecking", "1");
+        System.setProperty("org.apache.flink.kinesis.shaded.com.amazonaws.sdk.disableCbor", "1");
+    }
+
+    @Test(timeout = 120_000)
+    public void testTableApiSourceAndSink() throws Exception {
+        Properties properties = new Properties();
+        properties.setProperty(AWS_ENDPOINT, kinesalite.getEndpointUrl());
+
+        List<Order> smallOrders = ImmutableList.of(new Order("A", 5), new Order("B", 10));
+
+        // Large orders have a quantity > 10
+        List<Order> largeOrders =
+                ImmutableList.of(new Order("C", 15), new Order("D", 20), new Order("E", 25));
+
+        KinesisPubsubClient client = new KinesisPubsubClient(properties);
+        client.createTopic(ORDERS, 1, properties);
+        client.createTopic(LARGE_ORDERS, 1, properties);
+
+        smallOrders.forEach(order -> client.sendMessage(ORDERS, toJson(order)));
+        largeOrders.forEach(order -> client.sendMessage(ORDERS, toJson(order)));
+
+        executeSqlStatements(readSqlFile("filter-large-orders.sql"));
+
+        List<Order> result = readAllOrdersFromKinesis(client);
+        assertEquals(largeOrders.size(), result.size());
+        assertTrue(result.containsAll(largeOrders));
+    }
+
+    private List<Order> readAllOrdersFromKinesis(final KinesisPubsubClient client)
+            throws Exception {
+        Deadline deadline = Deadline.fromNow(Duration.ofSeconds(5));
+        List<Order> orders;
+        do {
+            Thread.sleep(1000);
+            orders =
+                    client.readAllMessages(LARGE_ORDERS).stream()
+                            .map(order -> fromJson(order, Order.class))
+                            .collect(Collectors.toList());
+        } while (deadline.hasTimeLeft() && orders.size() < 3);
+
+        return orders;
+    }
+
+    private List<String> readSqlFile(final String resourceName) throws Exception {
+        return Files.readAllLines(Paths.get(getClass().getResource(resourceName).toURI()));
+    }
+
+    private void executeSqlStatements(final List<String> sqlLines) throws Exception {
+        flink.submitSQLJob(
+                new SQLJobSubmission.SQLJobSubmissionBuilder(sqlLines)
+                        .addJars(sqlConnectorKinesisJar)
+                        .build());
+    }
+
+    @SneakyThrows
+    private <T> String toJson(final T object) {
+        return new ObjectMapper().writeValueAsString(object);
+    }
+
+    @SneakyThrows
+    private <T> T fromJson(final String json, final Class<T> type) {
+        return new ObjectMapper().readValue(json, type);
+    }
+
+    /** Test data model class for sending and receiving records on Kinesis. */
+    @Data
+    @EqualsAndHashCode
```

Comment:
`@EqualsAndHashCode` is implied by `@Data`

Reply:
Removed lombok